apps/remix-ide/src/app/files/dgitProvider.js

```diff
@@ -287,36 +294,58 @@ class DGitProvider extends Plugin {
     }
   };
 
+  async importIPFSFiles (config, cid, workspace) {
+    const ipfs = IpfsHttpClient(config)
+    let result = false
+    try {
+      const data = ipfs.get(cid, { timeout: 2000 })
```

Comment:
`2000` for the timeout might not be enough in some cases where the content hasn't been queried since long

Reply:
true. what would you advise?

Reply:
if the UX allows the user to wait for the response, i'd say 10000 is fine..